packages/remirror__extension-link/src/link-extension.ts

```diff
@@ -163,7 +163,7 @@ export type LinkAttributes = ProsemirrorAttributes<{
     defaultProtocol: '',
     selectTextOnClick: false,
     openLinkOnClick: false,
-    autoLinkRegex: /(http:\/\/www\.|https:\/\/www\.|http:\/\/|https:\/\/)?[\da-z]+([.-][\da-z]+)*\.[a-z]{2,8}(:\d{1,5})?(\/\S*)?/,
+    autoLinkRegex: /((?:http:\/\/www\.|https:\/\/www\.|http:\/\/|https:\/\/)?[\da-z]+(?:[.-][\da-z]+)*\.[a-z]{2,8}(?::\d{1,5})?(\/\S*)?)/,
```

Comment:
Does this work will full URL (including a path) as there is a capture group on `(\/\/S*)`

Reply:
I also think the beginning of this regex is a bit unnecessarily verbose.

I think we could just use

```suggestion
    autoLinkRegex: /(?:https?:\/\/)?[\da-z]+(?:[.-][\da-z]+)*\.[a-z]{2,8}(?::\d{1,5})?(?:\/\S*)?/gi,
```